admin/autoupdater/src/autoupdater.c

```diff
@@ -350,6 +350,8 @@ static bool autoupdate(const char *mirror, struct settings *s, int lock_fd) {
 		}
 	}
 
+	free_manifest_data(m);
```

Comment:
More important than freeing the manifest in the successful case would be to free it when false is returned, as that is the case where more mirrors are tried next.